### PR TITLE
Docs: Togroups getting started guide

### PR DESCRIPTION
**Pull request by eloycoto**
Two commits: 

- GSG for AWS
- Fix doc issues.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6340)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-docs-please

File: Documentation/gettingstarted/aws.rst [hunk 1]
```diff
@@ -0,0 +1,303 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+****************************************
+Getting Started Securing AWS instances
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: Instances

File: Documentation/gettingstarted/aws.rst [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+based on AWS instances metadata. It is a detailed walk-through of getting a
+single-node Cilium environment running on your machine. It is designed to take
+15-30 minutes with some experience running Kubernetes.
+
+.. include:: minikube_intro.rst
+
+Step 1: Create AWS secrets
+===========================
+
+Before install Cilium a new Kubernetes Secret with the AWS Tokens need to be in
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: Before installing Cilium, a new...

Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: need --> needs

File: Documentation/gettingstarted/aws.rst [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+place, that secret will allow Cilium to gather the info from Amazon AWS API
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: ... from the Amazon ...

File: Documentation/gettingstarted/aws.rst [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+and create a new **derived Cilium Network Policy** from it.
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: I'd remove this notion of "derived Cilium Network Policy". Remove everything after 'and'.

File: Documentation/gettingstarted/aws.rst [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+
+AWS Access keys and IAM role
+------------------------------
+
+To create a new access tokens the `following guide can be used
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: token

File: Documentation/gettingstarted/aws.rst [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+<https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-quick-configuration>`_,
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: Change the comma to a period.

File: Documentation/gettingstarted/aws.rst [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+these keys will need a specific permissions, Cilium needs the following:
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: These keys need to have certain permissions set:

File: Documentation/gettingstarted/aws.rst [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": "ec2:Describe*",
+                "Resource": "*"
+            }
+        ]
+    }
+
+As soon that you have the access tokens, the following secret need to be added,
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: As soon as you have

Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: needs to be added.

File: Documentation/gettingstarted/aws.rst [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+where the data need to be encoded in base64:
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: --> It must be encoded in base64.

File: Documentation/gettingstarted/aws.rst [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+      AWS_DEFAULT_REGION: ""
+
+To encode in base64, the following script can be used:
+
+.. parsed-literal::
+
+    $ echo -n "eu-west-1"  | base64
+    ZXUtd2VzdC0x
+
+This secret stores the AWS credentials and Cilium Operator will use that
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: ... stores the AWS credentials, which will be used to connect to the AWS API.

File: Documentation/gettingstarted/aws.rst [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+Where at least one pod should be available.
+
+
+Step 2: Create a sample policy
+================================
+
+Deploy a demo application:
+----------------------------
+
+In this case we're going to use a demo application that it is used in other
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: that it is used --> that is used

File: Documentation/gettingstarted/aws.rst [hunk 12]
```diff
@@ -0,0 +1,303 @@
...
+guides. Example manifest will create three microservices applications:
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: These manifests will create three microservices:

File: Documentation/gettingstarted/aws.rst [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+    po/xwing                         1/1       Running   0          1m
+
+    NAME             TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
+    svc/deathstar    ClusterIP   10.109.254.198   <none>        80/TCP    3h
+    svc/kubernetes   ClusterIP   10.96.0.1        <none>        443/TCP   3h
+
+Policy Language:
+-----------------
+
+To enforce security to AWS instances, the **ToGroups** rule will be used. This
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: I'd rephrase this since we currently only interface with AWS. What about: 

"**ToGroups** rules can be used to define policy in relation to cloud providers, like AWS."

File: Documentation/gettingstarted/aws.rst [hunk 14]
```diff
@@ -0,0 +1,303 @@
...
+      - toPorts:
+        - ports:
+          - port: '80'
+            protocol: TCP
+        toGroups:
+        - aws:
+            securityGroupsIds:
+            - 'sg-0fef46100a88d03c3'
+
+This policy allows traffic from pod *xwing* to any AWS instance that is on
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: on --> in

File: Documentation/gettingstarted/aws.rst [hunk 15]
```diff
@@ -0,0 +1,303 @@
...
+the Security group with ID ``sg-0fef46100a88d03c3``. Also the rule can be by
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: Security --> security

Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: I'd remove this sentence, since we don't use instance labels, security group names, or regions in this guide. We can document this specifically in the policy documentation for ToGroups.

File: Documentation/gettingstarted/aws.rst [hunk 16]
```diff
@@ -0,0 +1,303 @@
...
+instance labels, security groups names or region.
+
+
+Validate that derived policy is in place
+----------------------------------------
+
+Every time that a new policy with ToGroups rules is added a new derivative
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: This is a run-on sentence. I'd also change the 
"...is added, an equivalent policy (also called "derivative policy"), will be created. This policy will contain the set of CIDRs that correspond to the specification in ToGroups, e.g., the IPs of all instances that are part of a specified security group. "

Maybe specify that the policy is updated 'periodically' vs. 'every five minutes' - I'm worried that if this resync period is updated in the future that this documentation will fall out of sync with the codebase. This is nit-picky though :-)

File: Documentation/gettingstarted/aws.rst [hunk 17]
```diff
@@ -0,0 +1,303 @@
...
+policy will be created, the derivative policy will contain the list of allowed
+IPs in the ToCIDRSet. This new policy will be updated every five minutes to get
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: What about: "The list of IPs will be updated periodically." ?

File: Documentation/gettingstarted/aws.rst [hunk 18]
```diff
@@ -0,0 +1,303 @@
...
+AWS instances changes.
+
+.. parsed-literal::
+
+    $ kubectl get cnp
+    NAME                                                             AGE
+    to-groups-sample                                                 11s
+    to-groups-sample-togroups-044ba7d1-f491-11e8-ad2e-080027d2d952   10s
+
+Where the derivative policy has the following entries on the ToCIDR:
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: "Eventually, the derivative policy will contain IPs in the ToCIDR section:"

I'm unsure as to whether "ToCIDR" should be formatted in a specific way.

File: Documentation/gettingstarted/aws.rst [hunk 19]
```diff
@@ -0,0 +1,303 @@
...
+        value: CiliumNetworkPolicy
+    status:
+      nodes:
+        k8s1:
+          enforcing: true
+          lastUpdated: 2018-11-30T11:28:03.907678888Z
+          localPolicyRevision: 28
+          ok: true
+
+The important information that the derivative rule should have is:
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: The derivative rule should contain the following information:

File: Documentation/gettingstarted/aws.rst [hunk 20]
```diff
@@ -0,0 +1,303 @@
...
+
+- *metadata.OwnerReferences*: that contains the information about the parent
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: At the start of the guide, define what a parent policy and derivative policy are so users know the terminology.

File: Documentation/gettingstarted/aws.rst [hunk 21]
```diff
@@ -0,0 +1,303 @@
...
+  policy.
+
+- *specs.Egress.ToCIDRSet*: that is the list of the instances private and public
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: : the list of private and public IPs of the instances that correspond to the spec of the parent policy.

File: Documentation/gettingstarted/aws.rst [hunk 22]
```diff
@@ -0,0 +1,303 @@
...
+  IPs for the parent to groups filter.
+
+- *status*: that is the policy enforce status and when is updated.
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst]: whether or not the policy is enforced yet, and when the policy was last updated.

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/aws.rst [hunk 23]
```diff
@@ -0,0 +1,301 @@
...
+based on AWS instances metadata. It is a detailed walk-through of getting a
+single-node Cilium environment running on your machine. It is designed to take
+15-30 minutes with some experience running Kubernetes.
+
+.. include:: minikube_intro.rst
+
+Step 1: Create AWS secrets
+===========================
+
+Before install Cilium, a new Kubernetes Secret with the AWS Tokens needs to be
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst:21]: install --> installing

File: Documentation/gettingstarted/aws.rst [hunk 24]
```diff
@@ -0,0 +1,301 @@
...
+in place, that secret will allow Cilium to gather the info from the Amazon AWS
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst:22]: Run-on sentence.

"... needs to exist in your Kubernetes cluster. This secret will allow..."

File: Documentation/gettingstarted/aws.rst [hunk 25]
```diff
@@ -0,0 +1,301 @@
...
+API.
+
+AWS Access keys and IAM role
+------------------------------
+
+To create a new access token the `following guide can be used
+<https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-quick-configuration>`_.
+These keys need to have certain permission set:
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst:30]: permission --> permissions

File: Documentation/gettingstarted/aws.rst [hunk 26]
```diff
@@ -0,0 +1,301 @@
...
+                key: AWS_SECRET_ACCESS_KEY
+                optional: true
+          - name: AWS_DEFAULT_REGION
+            valueFrom:
+              secretKeyRef:
+                name: cilium-aws
+                key: AWS_DEFAULT_REGION
+                optional: true
+
+To list all the AWS instances available the following command can be used:
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst:113]: To list all of the available AWS instances, the following...

File: Documentation/gettingstarted/aws.rst [hunk 27]
```diff
@@ -0,0 +1,301 @@
...
+
+
+Step 2: Create a sample policy
+================================
+
+Deploy a demo application:
+----------------------------
+
+In this case we're going to use a demo application that is used in other guides.
+These manifest will create three microservices applications: *deathstar*,
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst:139]: manifest --> manifests

File: Documentation/gettingstarted/aws.rst [hunk 28]
```diff
@@ -0,0 +1,301 @@
...
+            securityGroupsIds:
+            - 'sg-0fef46100a88d03c3'
+
+This policy allows traffic from pod *xwing* to any AWS instance that is in
+the security group with ID ``sg-0fef46100a88d03c3``.
+
+Validate that derived policy is in place
+----------------------------------------
+
+Every time that a new policy with ToGroups rules is added,an equivalent policy
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst:204]: add a space after the comma: "added,an" --> "added, an"

File: Documentation/gettingstarted/aws.rst [hunk 29]
```diff
@@ -0,0 +1,301 @@
...
+(also called "derivative policy"), will be created. This policy will contain the
+set of CIDRs that correspond to the specification in ToGroups, e.g., the IPs of
+all instances that are part of a specified security group. Also the list of IPs
```
Comment (ianvernon) [inline Documentation/gettingstarted/aws.rst:207]: This list of IPs will be updated periodically.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Looks better, but still needs some grammatical fixes.

**Comment by ianvernon** [comment]
Superseded by #6417 .
